Documentation/WorkflowSwiftUI Adoption Guide.md

```diff
@@ -0,0 +1,448 @@
+        case increment
+
+        func apply(toState state: inout State) -> Never? {
+            state.counter += 1
+            return nil
+        }
+    }
+
+    func makeInitialState() -> State {
+        .init()
+    }
+
+    func render(
+        state: State,
+        context: RenderContext<Self>
+    ) -> ActionModel<State, Action> {
+        context.makeActionModel(state: state)
+    }
+}
+
+```
+
+For complex workflows that have multiple actions or compose observable models from child
+workflows, you can create a custom model that conforms to `ObservableModel`:
+
+```swift
+struct ComplexWorkflow: Workflow {
+    typealias Output = Never
+
+    @ObservableState
+    struct State {
+        var counter = 0
+    }
+
+    enum UpAction: WorkflowAction {
+        typealias WorkflowType = ComplexWorkflow
+        case increment
+
+        func apply(toState state: inout State) -> Never? {
+            state.counter += 1
+            return nil
+        }
+    }
+
+    enum DownAction: WorkflowAction {
+        typealias WorkflowType = ComplexWorkflow
+        case decrement
+
+        func apply(toState state: inout State) -> Never? {
+            state.counter -= 1
+            return nil
+        }
+    }
+
+    func makeInitialState() -> State {
+        .init()
+    }
+
+    func render(
+        state: State,
+        context: RenderContext<Self>
+    ) -> CustomModel {
+        CustomModel(
+            accessor: context.makeStateAccessor(state: state),
+            child: TrivialWorkflow().rendered(in: context),
+            up: context.makeSink(of: UpAction.self),
+            down: context.makeSink(of: DownAction.self)
+        )
+    }
+}
+
+struct CustomModel: ObservableModel {
+    var accessor: StateAccessor<ComplexWorkflow.State>
+
+    var child: TrivialWorkflow.Rendering
+
+    var up: Sink<ComplexWorkflow.UpAction>
+    var down: Sink<ComplexWorkflow.DownAction>
+}
+```
+
+## Views
+
+In your View, you’ll access state and sinks via a property of type `Store<Model>`. This type wraps the model that your workflow renders, and provides conveniences for access the state, sinks, and child stores.
+
+If targeting iOS 16 or below, you’ll need to wrap your view’s body in `WithPerceptionTracking`. This is a component of the `Perception` backport that allows observation to work on iOS before iOS 17.
+
+```swift
+struct PersonState {
+  var name: String
+  var age: Int
+}
+
+enum PersonAction {
+  case giveHighFive
+}
+
+typealias PersonModel = ActionModel<PersonState, PersonAction>
+
+struct PersonView: View {
+  let store: Store<PersonModel>
+  
+  var body: some View {
+    WithPerceptionTracking {
+      VStack {
+        Text("Name: \(store.name)")
+        Text("Age: \(store.age)")
+
+        Button("✋") {
+          store.send(.giveHighFive)
+        }
+      }
+    }
+  }
+}
+```
+
+If you forget to wrap your view's body in `WithPerceptionTracking`, a runtime warning will remind you.
```

Comment:
Should we turn this into an assert as well? I worry folks will miss the runtime warning. Not sure if this is possible though since I assume this comes from the point free library!

Reply:
The good news is that their runtime warning will fail tests when they are encountered in that context.